hoomd/hpmc/IntegratorHPMCMonoNEC.h

```diff
@@ -0,0 +1,1368 @@
+
+        // GlobalArray< hpmc_counters_t >     m_count_total;      // Inherited from base class
+        GlobalArray< hpmc_nec_counters_t > m_nec_count_total;  //!< counters for chain statistics
+
+//         unsigned long int count_moved_particles; // Counts translations that result in a collision (including the last particle)
+//         unsigned long int count_moved_again;     // Counts translations that do not result in a collision (or end of chain)
+//         unsigned long int count_move_attempts;   // Counts chains
+//         unsigned long int count_events;          // Counts everything (translations, repeated translations, and rotations [if applicable])
+//
+//         unsigned long int count_tuner_chains;
+//         unsigned long int count_tuner_collisions;
+
+        // statistics - pressure
+        // We follow the equations of Isobe and Krauth, Journal of Chemical Physics 143, 084509 (2015)
+        Scalar count_pressurevirial;
+        Scalar count_movelength;
+
+    private: // in line with IntegratorHPMC
+        hpmc_nec_counters_t m_nec_count_run_start;             //!< Count saved at run() start
+        hpmc_nec_counters_t m_nec_count_step_start;            //!< Count saved at the start of the last step
+
+    public:
+        //! Construct the integrator
+        IntegratorHPMCMonoNEC(std::shared_ptr<SystemDefinition> sysdef);
+        //! Destructor
+        virtual ~IntegratorHPMCMonoNEC();
+
+        //! Reset statistics counters
+        virtual void resetStats()
+            {
+            this->m_exec_conf->msg->notice(10) << "IntegratorHPMCMonoNEC<Shape>::resetStats" << std::endl;
+
+            IntegratorHPMCMono<Shape>::resetStats();
+
+            // Taking IntegratorHPMC.h as reference
+            ArrayHandle<hpmc_nec_counters_t> h_nec_counters(m_nec_count_total, access_location::host, access_mode::read);
+            m_nec_count_run_start = h_nec_counters.data[0];
+            }
+
+        //! Change move ratio
+        /*! \param m_chain_time set duration of a chain
+        */
+        void setChainTime(Scalar chain_time)
+            {
+            this->m_exec_conf->msg->notice(10) << "IntegratorHPMCMonoNEC<Shape>::setChainTime(" << chain_time << ")" << std::endl;
+            m_chain_time = chain_time;
+            }
+
+        //! Get move ratio
+        //! \returns ratio of translation versus rotation move attempts
+        inline Scalar getChainTime()
+            {
+            return m_chain_time;
+            }
+
+        //! Change move ratio
+        /*! \param m_chain_time set duration of a chain
+        */
+        void setChainProbability(Scalar chain_probability)
+            {
+            this->m_exec_conf->msg->notice(10) << "IntegratorHPMCMonoNEC<Shape>::setChainProbability(" << chain_probability << ")" << std::endl;
+            m_chain_probability = static_cast<unsigned int>(chain_probability * 65536.0);
```

Comment:
Is there a specific reasoning behind this specific factor?

Reply:
The chain-probability is the chain equivalent to translation-move-probability in local update HPMC and defined the same way.

https://github.com/glotzerlab/hoomd-blue/blob/96d089d297770d1fade62d9b6f7a60c7340cbd6e/hoomd/hpmc/IntegratorHPMC.h#L313-L316